### DISCUSSION TITLE
In Activity Output/Input, is it possible to select only Trasient as Storage?

### DISCUSSION THREAD

**Discussion by UserNameIsABC** [elsa-workflows/elsa-core#1492] [Q&A]
What is the Storage setting in the Activity?

In Activity Output/Input, I think we can choose the Storage from Default/Trasient/WorkflowInstance/Blob Storage.
Also, in Startup.cs, I think we can set the Default to the specified Storage by setting the following

`Services.AddElsa(elsa => elsa.UseDefaultWorkflowStorageProvider<TransientWorkflowStorageProvider>());`

In Activity Output/Input, is it possible to select only Trasient as Storage?
(I want to exclude other options.)

Also, if there is a way to set this for each Activity and each Output/Input, 
I would appreciate it if you could let me know how to do that as well.

**Comment by sfmskywalker** [comment]
Some background information about activity storage can be found [here](https://elsa-workflows.github.io/elsa-core/docs/next/activities/common#storage).

Currently all three storage providers are registered with DI by default (via `AddElsaCore` which is invoked with `AddElsa`), but I can see that it would be useful to control this from the application itself so I created [this issue](https://github.com/elsa-workflows/elsa-core/issues/1493) for it.

> Also, if there is a way to set this for each Activity and each Output/Input,
If you do `UseDefaultWorkflowStorageProvider<TransientWorkflowStorageProvider>`, then this is the default provider for all activities' Output/Input.

If there are more than one providers registered, then you can configure per activity property what provider to use (if you want to override the default one).

From the workflow designer, you can do this from the activity editor. For example:

![image](https://user-images.githubusercontent.com/938393/132487439-a8a2f9f7-cc63-4e00-8706-0f5742fbfc8e.png)


When writing workflows using C# and the workflow builder API, you can configure the provider to use like this:

![image](https://user-images.githubusercontent.com/938393/132487922-10e16724-4ec3-4cde-8ea0-b98eb069a49c.png)
